consensus/src/epoch_manager.rs

```diff
@@ -836,27 +841,43 @@ impl EpochManager {
             self.filter_quorum_store_events(peer_id, &unverified_event)?;
 
             // same epoch -> run well-formedness + signature check
-            let verified_event = monitor!(
-                "verify_message",
-                unverified_event.clone().verify(
-                    peer_id,
-                    &self.epoch_state().verifier,
-                    self.quorum_store_enabled
-                )
-            )
-            .context("[EpochManager] Verify event")
-            .map_err(|err| {
-                error!(
-                    SecurityEvent::ConsensusInvalidMessage,
-                    remote_peer = peer_id,
-                    error = ?err,
-                    unverified_event = unverified_event
-                );
-                err
-            })?;
-
-            // process the verified event
-            self.process_event(peer_id, verified_event)?;
+            let epoch_state = self.epoch_state.clone().unwrap();
+            let quorum_store_enabled = self.quorum_store_enabled;
+            let quorum_store_msg_tx = self.quorum_store_msg_tx.clone();
+            let buffer_manager_msg_tx = self.buffer_manager_msg_tx.clone();
+            let round_manager_tx = self.round_manager_tx.clone();
+            let my_peer_id = self.author;
+            self.bounded_executor
```

Comment:
Would we risk reordering messages when we forward the event after verification?

Reply:
bounded_executor is set to 2 threads, so this will reorder messages. 
where is that a problem?

Reply:
I set to 16 for production, I think most of our protocol tolerates reordering, the only part that doesn't work well is the fragments but we're going to remove it soon

Reply:
Hmm, if the batches are created every 30ms, for instance, there could be a lagging validator that might receive two batches with consecutive sequence numbers (say, 100 and 101) but the bounded_executor would reorder them as 101, 100.

Reply:
yeah that's possible, but the re-ordering is not likely to happen, 30ms is a long time compared to verification time (1-3ms), we can discuss whether we want to tolerate out of order batch id too